test/system/280-update.bats

```diff
@@ -161,4 +161,552 @@ device-write-iops   = /dev/zero:4000 | - | -
     run_podman rm -f -t0 testctr
 }
 
+function _create_container_check_change_of_HealthCheck_configuration_with_podman_update(){
+    local ctrname="$1"
+    local msg="$2"
+    local format="$3"
+    local flag="$4"
+    local value="$5"
+    local expect="$6"
+    local expect_msg="$7"
+    local expect_exit_code="$8"
+    local is_startup="$9"
+    local no_hc="${10}"
+
+    if [[ $is_startup = "yes" ]]; then
+        run_podman run -d --name $ctrname    \
+            --health-cmd "echo $msg"         \
+            --health-startup-cmd "echo $msg" \
+            $IMAGE /home/podman/pause
+        cid="$output"
+    else
+        if [[ $no_hc = "yes" ]]; then
+            run_podman run -d --name $ctrname \
+                    $IMAGE /home/podman/pause
+            cid="$output"
+        else
+            run_podman run -d --name $ctrname \
+                --health-cmd "echo $msg"  \
+                $IMAGE /home/podman/pause
+            cid="$output"
+        fi
+    fi
```

Comment:
there is a lot of duplication here, first you can use elif to avoid one layer of nesting

In general it is hard for the reader to figure out the differences, you should split out the run_podman call and only do the stuff that is different in there, i.e.  define a `hc_flags` arrray where you add the flags too and then to the run_podman call below which would make this a lot simpler.

Reply:
Second, what Paul said